lib/adagrams.rb

```diff
@@ -0,0 +1,136 @@
+#wave 1
+def draw_letters
+  letters_pool = []
+
+  letter_quantity = {"A": 9, "B": 2, "C": 2, "D": 4, "E": 12, "F": 2, "G": 3, "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6, "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 2, "U": 4, "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1}
+
+  letter_quantity.each do |letter, qty|
+    qty.times do
+      letters_pool << letter.to_s
+    end
+  end
+
+  #p letters_pool
+  drawn_letters = letters_pool.sample(10)
+  return drawn_letters
+end
+
+##################################################
+#Wave 2
+def uses_available_letters?(input, letters_in_hand)
+  valid_word = nil
+  input_as_array = input.upcase.split("")
+  hand = letters_in_hand.dup
+  letters_boolean = []
+
+  input_as_array.each do |letter|
+    if hand.include?(letter)
+      index = hand.index(letter)
+      hand.delete_at(index)
+      letters_boolean << true
+    else
+      letters_boolean << false
+    end
+  end
+
+  if letters_boolean.include?(false)
+    valid_word = false
+  else
+    valid_word = true
+  end
+
+  return valid_word
+end
+##################################################
+#wave 3
+def score_word(word)
+  score = 0
+  word_array = word.upcase.split("")
+
+  word_array.each do |letter|
+    case letter
+      when "A", "E", "I", "O", "U", "L", "N", "R", "S","T"
+        score += 1
+      when "D", "G"
+        score += 2
+      when "B", "C", "M", "P"
+        score += 3
+      when "F", "H", "V","W", "Y"
+        score += 4
+      when "K"
+        score += 5
+      when "J", "X"
+        score += 8
+      when "Q", "Z"
+        score += 10
+    end
+  end
+
+  if word_array.length >= 7 && word_array.length <=10
+    score += 8
+  end
+
+  return score
+end
+################################################
+#wave 4
+def highest_score_from(words)
+  array_of_words = []
+
+  words.each do |word|
+    array_of_words <<  {
+        word: word,
+        score: score_word(word),
+        length: word.length
+    }
+  end
+
+  highest_scoring_word = array_of_words.max_by{ |word| word[:score].to_i}
```

Comment:
`to_i` is redundant here since `score_word` already returns an Integer.

```suggestion
  highest_scoring_word = array_of_words.max_by{ |word| word[:score] }
```